knebind/solver/solver.go

```diff
@@ -277,8 +280,22 @@ func (s *solver) solve() (*assign, error) {
 		port2Intfs := make(map[interface{}][]interface{})
 		for dut, node := range dev2Node {
 			for port, intfs := range dev2Node2Port2Intfs[dut.(*opb.Device)][node.(*tpb.Node)] {
+
 				for _, i := range intfs {
-					port2Intfs[port] = append(port2Intfs[port], i)
+					if port.GetGroup() != "" {
+						if i.groupName == port.GetGroup() {
+							port2Intfs[port] = append(port2Intfs[port], i)
+						}
+					} else {
+						if i.groupName == "" {
+							// ports with no group get high preference
```

Comment:
Can you explain what "high preference" here means? What is the value of this?

Reply:
I think I get that it is a performance optimization, not necessary for correctness. That is your understanding?

Reply:
@greg-dennis high preference means,if someone doesn't provide any group for a port,   what i have seen ondatra try to map all possible interfaces to then given port, so all grouped and non grouped ports are valid , but in the list non-grouped ports will get prepend to the possible interface list

Reply:
Yes